aips/aip-116.md

```diff
@@ -326,47 +395,63 @@ A new npm package named `@aptos-labs/siwa` will be created to house utilities fo
 Below are the primary functions that will be provided as part of the `@aptos-labs/siwa` utilities:
 
 ```ts
+export type VerificationResult<T> =
+  | { valid: true; data: T }
+  | { valid: false; errors: string[] };
+
 /**
- * Create a SIWA plain text message from the `AptosSignInInput` (including required fields) following the ABNF format defined in the Sign in with Aptos
+ * Create a SIWA Message from the `AptosSignInInput` with required fields following the ABNF format defined in the Sign in with Aptos
  * specifications.
  *
- * @param input The input fields to create the SIWA plain text message from.
+ * @param input The `AptosSignInInput` with required fields to create the SIWA message from.
  *
- * @returns The SIWA plain text message.
+ * @returns The SIWA message.
  */
-function createSignInMessagePlainText(
+function createSignInMessageText(
   input: AptosSignInInput & AptosSignInRequiredFields
```

Comment:
I think this `AptosSignInRequiredFields` appears out of nowhere and is not explained...